packages/fcl-ethereum-provider/src/network/network-manager.ts

```diff
@@ -0,0 +1,88 @@
+import {ACCESS_NODE_API_KEY, FLOW_CHAINS, FlowNetwork} from "../constants"
+import {
+  BehaviorSubject,
+  concat,
+  distinctUntilChanged,
+  filter,
+  firstValueFrom,
+  from,
+  map,
+  of,
+  Subscription,
+  switchMap,
+} from "../util/observable"
+import * as fcl from "@onflow/fcl"
+
+type ChainIdStore = {
+  isLoading: boolean
+  chainId: number | null
+  error: unknown | null
+}
+export class NetworkManager {
+  private $chainIdStore = new BehaviorSubject<ChainIdStore>({
+    isLoading: true,
+    chainId: null,
+    error: null,
+  })
+
+  constructor(config: typeof fcl.config) {
+    // Map FCL config to behavior subject
+    const $config = new BehaviorSubject<Record<string, unknown> | null>(null)
+    config.subscribe((cfg, err) => {
+      if (err) {
+        $config.error(err)
```

Comment:
Are you supposed to throw errors in observables vs do something like `$config.next({ error: err })`? Do you want it to terminate?